SSW.DetectService.js

```diff
@@ -28,43 +33,74 @@ function ServiceForDetection(name, favicon) {
-        console.log('Detect fail for ' + service.name);
+    var timer = window.setTimeout(function () {
+        //console.log('Detect fail for ' + service.name);
         image.src = '';
         image.onload = null;
         image.onerror = null;
         service.testComplete = true;
         $(document).trigger("DetectServiceComplete", service);
     }, 1500);
-    image.onload = function() {
+    image.onload = function () {
         clearTimeout(timer);
-        console.log('Detect pass for ' + service.name);
+        //console.log('Detect pass for ' + service.name);
         service.successful = true;
         service.testComplete = true;
         $(document).trigger("DetectServiceComplete", service);
     };
     image.src = service.favicon;
 }
 
+function DoDetectServiceBasedOnCountryCode(countryCode) {
+    //console.log('DetectBasedOnCountry: ', countryCode);
+    //console.log('User based in China: ', countryCode.toLowerCase() === 'cn');
+    for (var service in serviceDetector) {
+        if (window.serviceDetector.hasOwnProperty(service)) {
+            window.serviceDetector[service].testComplete = true;
+            if (countryCode.toLowerCase() === 'cn') {
+                window.serviceDetector[service].successful = service.toLowerCase() === 'youku' ? true : false;                
+            } else {                
+                window.serviceDetector[service].successful = true;
+            }
+            $(document).trigger("DetectServiceComplete", service);
+        }
+    }
+    //console.log('window.ServiceDetector object after: ', window.serviceDetector);
+}
+
+function isAppleMobileDevice() {
+    var match = /iphone|ipad|ipod/i.test(navigator.userAgent.toLowerCase());
+    //console.log('isAppleMobileDevice: ', match);
+    return match;
+}
 
-$(document).ready(function() {
+$(document).ready(function () {
 
     window.serviceDetector = {
-        youtube:   new ServiceForDetection("youtube", "https://youtube.com/favicon.ico"),
-        google:    new ServiceForDetection("google", "https://google.com/favicon.ico"),
-        facebook: new ServiceForDetection("facebook", "https://facebook.com/favicon.ico"),
-        youku: new ServiceForDetection("youku", "https://static.youku.com/v1.0.166/index/img/favicon.ico")
+        vimeo: new ServiceForDetection("vimeo", "https://vimeo.com/favicon.ico"),
+        youtube: new ServiceForDetection("youtube", "https://youtube.com/favicon.ico"),
+        youku: new ServiceForDetection("youku", "https://static.youku.com/v1.0.166/index/img/favicon.ico"),
+        google: new ServiceForDetection("google", "https://www.google.com/favicon.ico"),
+        facebook: new ServiceForDetection("facebook", "https://facebook.com/favicon.ico")
     };
 
-    for (var service in window.serviceDetector) {
-        if (window.serviceDetector.hasOwnProperty(service)) {
-            DoDetectService(window.serviceDetector[service]);
+    if (isAppleMobileDevice()) {

```

Comment:
Would it be better to check the country first and than do `DoDetectService`?
This way we can avoid calling things we know they are going to fail based on the country.